common/src/main/java/org/opensearch/ml/common/connector/HttpConnector.java

```diff
@@ -324,13 +324,38 @@ public <T> T createPayload(String action, Map<String, String> parameters) {
             payload = fillNullParameters(parameters, payload);
             StringSubstitutor substitutor = new StringSubstitutor(parameters, "${parameters.", "}");
             payload = substitutor.replace(payload);
-
             if (!isJson(payload)) {
-                throw new IllegalArgumentException("Invalid payload: " + payload);
+                String payloadAfterEscape = connectorAction.get().getRequestBody();
+                Map<String, String> escapedParameters = escapeMapValues(parameters);
```

Comment:
If payload is wrong will escape all parameters again. Escape good parameters again may generate wrong result. For example, the original prompt is "please replace all \n of the context with --", if we escape again , it becomes "please replace all \\n of the context with --"

Reply:
if a string is already partial escape, meaning the string escape for the next line symbol but doesn't escape for double quote, this change will still not fix the payload. Please let me know if there is better way to identify partial escape cases.